trunk/SUAVE/Input_Output/OpenVSP/vsp_read_propeller.py

```diff
@@ -0,0 +1,192 @@
+    # Propeller orientation
+    prop.orientation_euler_angles 	= [0.0,0.0,0.0]
+    prop.orientation_euler_angles[0] 	= vsp.GetParmVal(prop_id, 'X_Rotation', 'XForm') * Units.degrees
+    prop.orientation_euler_angles[1] 	= vsp.GetParmVal(prop_id, 'Y_Rotation', 'XForm') * Units.degrees
+    prop.orientation_euler_angles[2] 	= vsp.GetParmVal(prop_id, 'Z_Rotation', 'XForm') * Units.degrees
+
+    # Get the propeller parameter IDs
+    parm_id    = vsp.GetGeomParmIDs(prop_id)
+    parm_names = []
+    for i in range(len(parm_id)):
+        parm_name = vsp.GetParmName(parm_id[i])
+        parm_names.append(parm_name)
+
+    # Run the vsp Blade Element analysis
+    vsp.SetStringAnalysisInput( "BladeElement" , "PropID" , (prop_id,) )
+    rid = vsp.ExecAnalysis( "BladeElement" )
+    Nc  = len(vsp.GetDoubleResults(rid,"YSection_000"))
+    prop.number_points_around_airfoil = 2*Nc
+    
+    prop.CLi			    = vsp.GetParmVal(parm_id[parm_names.index('CLi')])
+    prop.blade_solidity 	    = vsp.GetParmVal(parm_id[parm_names.index('Solidity')])
+    prop.number_of_blades           = int(vsp.GetParmVal(parm_id[parm_names.index('NumBlade')]))
+    
+    prop.tip_radius                 = vsp.GetDoubleResults(rid, "Diameter" )[0] / 2 * units_factor
+    prop.radius_distribution        = np.array(vsp.GetDoubleResults(rid, "Radius" )) * prop.tip_radius
+    prop.radius_distribution[-1]    = 0.99 * prop.tip_radius # BEMT requires max nondimensional radius to be less than 1.0
+    prop.hub_radius 		    = prop.radius_distribution[0]
+    
+    prop.chord_distribution         = np.array(vsp.GetDoubleResults(rid, "Chord" ))  * prop.tip_radius # vsp gives c/R
+    prop.twist_distribution         = np.array(vsp.GetDoubleResults(rid, "Twist" ))  * Units.degrees
+    prop.sweep_distribution 	    = np.array(vsp.GetDoubleResults(rid, "Sweep" ))
+    prop.mid_chord_alignment        = np.tan(prop.sweep_distribution*Units.degrees)  * prop.radius_distribution
+    prop.thickness_to_chord         = np.array(vsp.GetDoubleResults(rid, "Thick" ))
+    prop.max_thickness_distribution = prop.thickness_to_chord*prop.chord_distribution * units_factor
+
+    prop.Cl_distribution            = np.array(vsp.GetDoubleResults(rid, "CLi" ))
+
+    number_of_radial_stations        = len(prop.chord_distribution)
+    
+    # Extra data from VSP BEM that SUAVE might not use...
+    prop.beta34 		= vsp.GetDoubleResults(rid, "Beta34" )[0]  # pitch at 3/4 radius
+    prop.pre_cone 		= vsp.GetDoubleResults(rid, "Pre_Cone")[0]
+    prop.rake 			= np.array(vsp.GetDoubleResults(rid, "Rake"))
+    prop.skew 			= np.array(vsp.GetDoubleResults(rid, "Skew"))
+    prop.axial 			= np.array(vsp.GetDoubleResults(rid, "Axial"))
+    prop.tangential 		= np.array(vsp.GetDoubleResults(rid, "Tangential"))
+
+    # Set prop rotation
+    prop.rotation = 1
+    
+    # ---------------------------------------------
+    # Rotor Airfoil
+    # ---------------------------------------------
+    if write_airfoil_file:
```

Comment:
Just let it default to SUAVE's DAE-51.

I realize the DAE-51 airfoil and fit isn't perfect. But the upside of assuming the NACA 4412 is minimal in comparison to the downside of having us distribute polars.

Reply:
Fixed